fastchat/model/model_adapter.py

```diff
@@ -292,6 +293,40 @@ def remove_parent_directory_name(model_path):
     return model_path.split("/")[-1]
 
 
+class PeftModelAdapter:
+    """Loads any "peft" model and it's base model."""
+
+    def match(self, model_path: str):
+        """Accepts any model path with "peft" in the name"""
+        return "peft" in model_path
+
+    def load_model(self, model_path: str, from_pretrained_kwargs: dict):
+        """Loads the base model then the (peft) adapter weights"""
+        config = PeftConfig.from_pretrained(model_path)
+        base_model_path = config.base_model_name_or_path
+        if "peft" in base_model_path:
+            raise ValueError(
+                f"PeftModelAdapter cannot load a base model with 'peft' in the name: {config.base_model_name_or_path}"
+            )
+
+        base_adapter = get_model_adapter(base_model_path)
+        base_model, tokenizer = base_adapter.load_model(
+            base_model_path, from_pretrained_kwargs
+        )
+        model = PeftModel.from_pretrained(model, model_path)
```

Comment:
`UnboundLocalError: local variable 'model' referenced before assignment`

did you mean `model = PeftModel.from_pretrained(base_model, model_path)`?

Reply:
Ah shoot I missed that.  Yes that's what should be there.  I wasn't expecting this to get merged by someone else.

Reply:
Here's a fix: https://github.com/lm-sys/FastChat/pull/1819